Swiftification/Dictionary.swift

```diff
@@ -56,6 +56,17 @@ public extension Dictionary {
         return result
     }
     
+    /// Unions the other dictionaries with `self`. If more than
+    /// one dictionary has the same key, it will take on the rightmost
+    /// dictionary's value.
+    mutating func formUnion(first: Dictionary, _ rest: Dictionary...) {
```

Comment:
@stephanheilner: I went to the API guidelines to try to find a better name to suggest, and it turns out `union` and `formUnion` is an actual example they give on how to "correctly" name things: https://swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage.

Given that it follows the guidelines, I think we should keep it. I think we'll get used to that style of naming.